packages/rainbowkit/src/components/ChainModal/Chain.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { Fragment, useContext } from 'react';
+import { useSwitchChain } from 'wagmi';
+import { isMobile } from '../../utils/isMobile';
+import { AsyncImage } from '../AsyncImage/AsyncImage';
+import { AsyncImageSrc } from '../AsyncImage/useAsyncImage';
+import { Box, BoxProps } from '../Box/Box';
+import { MenuButton } from '../MenuButton/MenuButton';
+import { I18nContext } from '../RainbowKitProvider/I18nContext';
+import { useRainbowKitChains } from '../RainbowKitProvider/RainbowKitChainContext';
+import { Text } from '../Text/Text';
+
+interface ChainProps {
+  chainId: number;
+  currentChainId: number;
+  switchChain: ReturnType<typeof useSwitchChain>['switchChain'];
+  chainIconSize: BoxProps['height'];
+  name: string | undefined;
+  isLoading: boolean;
+  iconBackground: string | undefined;
+  src: string | AsyncImageSrc | undefined | null;
+  idx: number;
+}
+
+const Chain = ({
+  chainId,
+  currentChainId,
+  switchChain,
+  chainIconSize,
+  isLoading,
+  src,
+  name,
+  iconBackground,
+  idx,
+}: ChainProps) => {
```

Comment:
I created `Chain` component so it will be more readable for us & for other people since `ChainModal` was one of the components that required a good amount of changes. Made sure to seperate components here a bit 👍

Reply:
Noice 👌